src/store/flow/flow.ts

```diff
@@ -342,20 +339,23 @@ export const actions: ActionTree<IFlowsState, IRootState> = {
    * @param flowId
    * @param resourceUuids
    */
-  flow_removeResourcesAndRelatedValidationsOnActiveFlow({state, commit, getters, rootGetters}, {resourceUuids}: {resourceUuids: IResource['uuid'][] }) {
-    const flow: IFlow = getters.activeFlow
-    const nonOrphanResourceUuids: IResource['uuid'][] = getters.nonOrphanResourceUuidsOnActiveFlow ?? []
+  async flow_removeResourcesAndRelatedValidationsOnActiveFlow(
+    {commit, getters},
+    {resourceUuids}: {resourceUuids: IResource['uuid'][]},
+  ): Promise<void> {
     console.debug('trying to clean those resources:', resourceUuids)
-    const nonOrphanResourceUuidsToBeDeleted = intersection(nonOrphanResourceUuids, resourceUuids)
-    if (!isEmpty(nonOrphanResourceUuidsToBeDeleted)) {
-      throw new Error(`Unsafe action: trying to delete non orphaned resources ${nonOrphanResourceUuidsToBeDeleted}`)
-    }
+
+    const flow: IFlow = getters.activeFlow
     flow.resources = flow.resources.filter(item => !resourceUuids.includes(item.uuid))
     resourceUuids.forEach(
       uuid => commit('validation/removeValidationStatusesFor', {key: `resource/${uuid}`}, {root: true}),
     )
   },
```

Comment:
This action is used when deleting a choice (we don't check if it's orphan because the block still exists)